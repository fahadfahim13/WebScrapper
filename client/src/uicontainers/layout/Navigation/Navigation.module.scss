$navigation-background-color: #7024c4;
$text-color: white;

.menuContainer {
  width: 50%;
  background-color: $navigation-background-color;
  border-bottom-color: $navigation-background-color;
  display: flex;
  align-items: center;
  justify-content: flex-end;
}

.navigationContainer {
  background-color: $navigation-background-color;
  width: 100%;
  height: 60px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 20px;
  color: $text-color;
}

.logoContainer {
  width: 20%;
  height: 30px;
  display: flex;
  align-items: center;
}
.buttonsContainer {
  display: flex;
  width: 30%;
  justify-content: space-evenly;
}

.whiteButton {
  color: #fff;
}

.navButton {
  color: $text-color;
  background-color: $navigation-background-color;
  margin-right: 5%;
  border: none;
}

.loginButton {
  @extend .navButton;
  border: 1px solid $text-color;
  display: flex;
  align-items: center;
}
