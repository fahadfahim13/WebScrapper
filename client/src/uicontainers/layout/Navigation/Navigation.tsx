import React from 'react';
import { Menu } from 'antd';
import styles from './Navigation.module.scss';

const Navigation = () => {
  return <Menu mode="horizontal" className={styles.menuContainer} />;
};

export default Navigation;
