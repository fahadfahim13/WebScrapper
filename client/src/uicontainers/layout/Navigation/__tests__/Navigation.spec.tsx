import React from 'react';
import { create } from 'react-test-renderer';
import Navigation from '../Navigation';

jest.mock('antd', () => ({
  ...jest.requireActual('antd'),
  Row: 'Row',
  Col: 'Col',
  Button: 'Button',
  Image: 'Image',
  Menu: 'Menu',
}));

jest.mock('@ant-design/icons', () => ({
  ...jest.requireActual('@ant-design/icons'),
  DownOutlined: 'DownOutlined',
  PlusOutlined: 'PlusOutlined',
}));

describe('Navigation Page Testing', () => {
  describe('Navigation Bar Testing', () => {
    it('should match navigation bar snapshot', () => {
      const navigationTree = create(<Navigation />).toJSON();
      expect(navigationTree).toMatchSnapshot();
    });
  });
});
