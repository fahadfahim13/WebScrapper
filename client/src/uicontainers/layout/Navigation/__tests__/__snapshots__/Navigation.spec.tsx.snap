// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Page Testing Navigation Bar Testing should match navigation bar snapshot 1`] = `
<Menu
  className="menuContainer"
  mode="horizontal"
/>
`;
